boxoffice/models/line_item.py

```diff
@@ -215,7 +216,17 @@ def get_from_item(cls, item, qty, coupon_codes=[]):
         coupon_policies = item.discount_policies.filter(DiscountPolicy.discount_type == DISCOUNT_TYPE.COUPON).all()
         coupon_policy_ids = [cp.id for cp in coupon_policies]
         for coupon_code in coupon_codes:
-            coupon = DiscountCoupon.query.filter(DiscountCoupon.discount_policy_id.in_(coupon_policy_ids), DiscountCoupon.code == coupon_code).one_or_none()
+            if DiscountPolicy.is_signed_code_format(coupon_code):
+                policy = DiscountPolicy.get_from_signed_code(coupon_code)
+                if policy and policy.id in coupon_policy_ids:
+                    coupon = DiscountCoupon.query.filter_by(discount_policy=policy, code=coupon_code).one_or_none()
+                    if not coupon:
+                        coupon = DiscountCoupon(id=uuid.uuid4(), discount_policy=policy, code=coupon_code, usage_limit=1, used_count=0)
```

Comment:
Use `uuid1mc` not `uuid4`. This should be in DiscountCoupon's constructor, not here.